src/core/load-fragment.c

```diff
@@ -61,6 +61,19 @@
 #include "utf8.h"
 #include "web-util.h"
 
+static int socket_protocol_from_name_supported(const char *s) {
+        int r;
+
+        r = socket_protocol_from_name(s);
```

Comment:
hmm, after having a look at the code of socket_protocol_from_name() it appears it currently returns "0" in case of an unknown protocol. You check for < 0 here though, and return it as if it was a negative errno value.

Now, most of our `xyz_from_string()` calls generally return -1 on error, in the expectation that the underlying enum has some `xyz_INVALID` value defined to -1.

I think there's a bit of clean-up to do here really, I figure socket_protocol_from_name() should return -1 if it doesn't know the string, and I figure it should be renamed to socket_protocol_from_string() to match the naming we otherwise use for this kind of thing.

With that change in mind I figure this function here should be renamed parse_supported_socket_protocol() or so, to clarify it returns negative `errno` values rather than "-1" on failure... and of course it needs to be fixed to convert < 0 from socket_protocol_from_string() to -EINVAL or so